inference-engine/tests/functional/inference_engine/ov_tensor.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <cpp/ie_infer_request.hpp>
+#include <openvino/runtime/tensor.hpp>
+
+using namespace ::testing;
+using namespace std;
+using namespace InferenceEngine;
+using namespace InferenceEngine::details;
+
+
+TEST(TensorOVTests, throwsOnUninitializedData) {
+    ov::runtime::Tensor tensor;
+    ASSERT_THROW(tensor.data(), InferenceEngine::NotAllocated);
+}
+
+TEST(TensorOVTests, throwsOnUninitializedAs) {
+    ov::runtime::Tensor tensor;
+    ASSERT_THROW(tensor.as<float>(), InferenceEngine::NotAllocated);
+}
+
+TEST(TensorOVTests, throwsOnAsWithIncorrectElementType) {
+    ov::runtime::Tensor tensor{ov::element::f32, {1, 1, 2, 2}};
+    ASSERT_THROW(tensor.as<std::int32_t>(), InferenceEngine::ParameterMismatch);
+}
+
+TEST(TensorOVTests, throwsOnUninitializedGetElementType) {
+    ov::runtime::Tensor tensor;
+    ASSERT_THROW(tensor.get_element_type(), InferenceEngine::NotAllocated);
+}
+
+TEST(TensorOVTests, throwsOnUninitializedGetShape) {
+    ov::runtime::Tensor tensor;
+    ASSERT_THROW(tensor.get_shape(), InferenceEngine::NotAllocated);
+}
```

Comment:
In general I don't see any valid test for new API:
1. We don't test that ov::Function compiled with new API and new methods of infer request work correctly
2. We don't have tests for `Tensor` itself, only invalid tests that check exceptions